src/select/__tests__/select.test.js

```diff
@@ -302,28 +308,29 @@ describe('Stateless select', function() {
         .find(StyledInput)
         .first()
         .find('input');
+
       input.simulate('change', {target: {value: 'a'}});
-      onTextInputChangePromise.then(() =>
-        expect(wrapper.find(StyledOption)).toHaveLength(3),
-      );
+      await sleep(20);
+      wrapper.update();
+      expect(wrapper.find(StyledOption)).toHaveLength(3);
 
       input.simulate('change', {target: {value: 'aa'}});
-      onTextInputChangePromise.then(() =>
-        expect(wrapper.find(StyledOption)).toHaveLength(2),
-      );
+      await sleep(20);
+      wrapper.update();
+      expect(wrapper.find(StyledOption)).toHaveLength(2);
 
       input.simulate('change', {target: {value: 'aaa'}});
-      onTextInputChangePromise.then(() =>
-        expect(wrapper.find(StyledOption)).toHaveLength(1),
-      );
+      await sleep(20);
+      wrapper.update();
+      expect(wrapper.find(StyledOption)).toHaveLength(1);
 
       input.simulate('change', {target: {value: 'aaaa'}});
-      onTextInputChangePromise.then(() =>
-        expect(wrapper.find(StyledOption)).toHaveLength(0),
-      );
+      await sleep(20);
+      wrapper.update();
+      expect(wrapper.find(StyledOption)).toHaveLength(0);
```

Comment:
These tests were essentially broken on master. They used promises without using jest's helpers for async tests, so the test would "succeed" but then later have failures:

![image](https://user-images.githubusercontent.com/875591/46427029-136d0880-c6f5-11e8-928b-7684e14ac7af.png)

Reply:
did you fixed this in your PR, or should we address it in a new one?

Reply:
oh I see, never mind

Reply:
thanks for fixing it